config/nrfconnect/chip-gn/build_overrides/chip.gni

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+declare_args() {
+  # Root directory for CHIP.
+  chip_root = rebase_path("//../../..")
```

Comment:
We should avoid paths like ("//.."). Such a path means that the root is incorrect because it does not contain all of the code. Absolute paths make caching less effective.

I might rather entirely remove stuff building in subdirectories than use absolute paths. Building in subdirectories was meant to provide an example.. this doesn't set a great one.

It might be OK if we restrict it to Windows and keep the symlinks and relative paths for other systems. Or we could do away with the separate .gn files and just script the CI system to create a new repository with CHIP as a submodule and test that it builds. Such an approach would work on Windows without contortions like /..

Reply:
Changed approach to build from the chip root. So we can resolve this comment.